gix-blame/src/file/function.rs

```diff
@@ -322,15 +327,121 @@ fn tree_diff_at_file_path(
     let tree_iter = odb.find_tree_iter(&tree_id, rhs_tree_buf)?;
     stats.trees_decoded += 1;
 
-    let mut recorder = gix_diff::tree::Recorder::default();
-    gix_diff::tree(parent_tree_iter, tree_iter, state, &odb, &mut recorder)?;
+    let mut recorder = Recorder::new(file_path.into());
+    // TODO
+    // `recorder` cancels the traversal by returning `Cancel` when a change to `file_path` is
+    // found. `gix_diff::tree` converts `Cancel` into `Err(...)` which is why we ignore its return
+    // value here. I don’t know whether this has the potential to hide bugs.
+    let _ = gix_diff::tree(parent_tree_iter, tree_iter, state, &odb, &mut recorder);
```

Comment:
Indeed, this is where one would have to ignore the `Cancelled` variant of the error, but fail on all others.
The reason this is an error is… that it's intended to be used for cancellation, which is when one wants this to error out easily.
Here this isn't too useful though.

Reply:
Thanks for the context, I’ve adapted this part.